logic.js

```diff
@@ -150,24 +150,26 @@ module.exports = {
         LOG_LOGIC('Task: Contract Deployment / Create Transaction')
         // take the sha256 hash of address+nonce, then extract the rightmost 20 bytes
         const contractAddr = computeContractAddr(senderAddress)
-
-        // @dev: currently, the gas cost is the gaslimit. This WILL change in the future
-        const gasAndAmount = payload.amount + payload.gasLimit
+        const gasAndAmount = payload.amount + payload.gasLimit;
+        
         if (!walletCtrl.sufficientFunds(senderAddress, gasAndAmount)) {
           LOG_LOGIC('Insufficient funds. Returning error to client.')
           throw new Error('Insufficient funds')
         }
         LOG_LOGIC(`Contract will be deployed at: ${contractAddr}`)
 
-        const nextAddr = await scillaCtrl.executeScillaRun(
+        const responseData = await scillaCtrl.executeScillaRun(
           payload,
           contractAddr,
           dir,
           currentBNum,
           payload.gasLimit
         )
         // Deduct funds
-        walletCtrl.deductFunds(senderAddress, payload.amount + payload.gasLimit)
+        let nextAddr = responseData.nextAddress;
+        let gasConsumed = payload.gasLimit - responseData.gasRemaining;
```

Comment:
Currently, pending confirmation from @AmritKumar, the I have fixed gas price at the ratio of 1:1 gas to zil. I think it is worth assigning a dummy `gasPrice`, such that the call will become `walletCtrl.deductFunds(senderAddress, payload.amount + gasConsumed * gasPrice)` to avoid having to refactor when gas price becomes a factor on the main net.